code/modules/vore/resizing/resize_vr.dm

```diff
@@ -180,82 +180,94 @@ var/const/RESIZE_A_SMALLTINY = (RESIZE_SMALL + RESIZE_TINY) / 2
 				return 1
-			if(src.m_intent == "walk") //Most likely intentionally stepping on them.
-				var/size_damage_multiplier = (src.size_multiplier - tmob.size_multiplier)
-				var/damage = (rand(15,30)* size_damage_multiplier) //Since stunned is broken, let's do this. Rand 15-30 multiplied by 1 min or 1.75 max. 15 holo to 52.5 holo, depending on RNG and size differnece.
+			else // Most likely intentionally stepping on them.
+				var/size_damage_multiplier = 0.25 * (src.size_multiplier / tmob.size_multiplier)
+				var/damage = (rand(15,30)* size_damage_multiplier) //Since stunned is broken, let's do this. Rand 15-30 multiplied by 1 min or 2 max. 15 holo to 60 holo, depending on RNG and size differnece.
 				tmob.apply_damage(damage, HALLOSS)
-				tmob.resting = 1
 				var/mob/living/carbon/human/H = src
+				log_and_message_admins("has stomped on [tmob], pinning them to the ground and dealing [damage] HALLOSS.") //Both humans and mobs, since stepping on mobs can be abused.
 				admin_attack_log(src, tmob, "Pinned [tmob.name] under foot for [damage] HALLOSS.", "Was pinned under foot by [src.name] for [damage] HALLOSS.", "Pinned [tmob.name] under foot for [damage] HALLOSS.")
 				if(istype(H) && istype(H.tail_style, /datum/sprite_accessory/tail/taur/naga))
-					src << "You push down on [tmob] with your tail, pinning them down under you!"
-					tmob << "[src] pushes down on you with their tail, pinning you down below them!"
+					src << "<span class='warning'>You push down on [tmob] with your tail, pinning them down under you!</span>"
+					tmob << "<span class='warning'>[src] pushes down on you with their tail, pinning you down below them!</span>"
 				else
-					src << "You firmly push your foot down on [tmob], painfully but harmlessly pinning them to the ground!"
-					tmob << "[src] firmly pushes their foot down on you, quite painfully but harmlessly pinning you do to the ground!"
-
+					src << "<span class='warning'>You firmly push your foot down on [tmob], painfully but harmlessly pinning them to the ground!</span>"
+					tmob << "<span class='warning'>[src] firmly pushes their foot down on you, quite painfully but harmlessly pinning you to the ground!</span>"
+				return 1
 
-	if(src.a_intent == I_HURT && src.canmove && !src.buckled)
-		if((src.get_effective_size() - tmob.get_effective_size()) >= 0.75)
-			now_pushing = 0
+	else if(src.a_intent == I_HURT)
+		if((src.get_effective_size() / tmob.get_effective_size()) >= 2)
 			src.forceMove(tmob.loc)
-			var/size_damage_multiplier = (src.size_multiplier - tmob.size_multiplier)
-			var/damage = (rand(1,3)* size_damage_multiplier) //Rand 1-3 multiplied by 1 min or 1.75 max. 1 min 5.25 max damage to each limb.
-			var/calculated_damage = damage/2 //This will sting, but not kill. Does .5 to 2.625 damage, randomly, to each limb.
+			now_pushing = 0
+			tmob.Weaken(4)
+			var/size_damage_multiplier = 0.25 * (src.size_multiplier / tmob.size_multiplier)
```

Comment:
Same as above.

Reply:
As above, your calculated values for maximum and minimum damage are wrong.

Reply:
Fixed them.